core/include/detray/propagator/actor_chain.hpp

```diff
@@ -0,0 +1,96 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include <type_traits>
+#include <utility>
+
+#include "detray/definitions/detail/accessor.hpp"
+#include "detray/definitions/qualifiers.hpp"
+
+namespace detray {
+
+/// The interface to the actors and aborters in the propagation.
+///
+/// It can hold both simple actors, as well as an actor with its observers.
+/// The states of the actors need to be passed to the chain in an external tuple
+///
+/// @tparam tuple_t tuple type used to resolve the actor types
+/// @tparam actors_t the types of the actors in the chain.
+template <template <typename...> class tuple_t = dtuple, typename... actors_t>
+class actor_chain {
+
+    public:
+    /// Types of the actors that are registered in the chain
+    using actor_list_type = tuple_t<actors_t...>;
+
```

Comment:
Can we define `actor_state` here?
`using actor_state = tuple_t<actors_t::state ...>`

Reply:
Done